site/docs/build-event-protocol.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: documentation
+title: Build Event Protocol
+---
+
+# Build Event Protocol
+
+The [Build Event Protocol] allows third party programs to gain insight into a Bazel invocation. The protocol is a set of [protocol buffer] messages with some semantics defined on top of it. It includes information about build and test results, build progress, the build configuration and much more. The Build Event Protocol is intended to be consumed programmatically and makes parsing Bazel’s command line output a thing of the past. For example, using the information from the Build Event Protocol one could create a dashboard that for a given build displays the targets built, tests run and artifacts generated. Another example would be an IDE plugin gathering information about the build targets and tests being executed.
+
+## The Build Event Protocol from 10,000 feet
+As mentioned before, the [Build Event Protocol] represents information about a build as events. A build event is a protocol buffer message consisting of a build event identifier, a set of child events and a payload.
+ 
+* __Build Event Identifier:__
+    Depending on the kind of build event, it might be an [opaque string] or [structured information] revealing more about the build event. A build event identifier is unique within a build.
+
+* __Children:__ A build event may announce other build events, by including their build event identifiers in its [children field]. For example, the `PatternExpanded` build event announces the targets it expands to as children. The main purpose of child events is to ensure that no event except for the initial event is posted unexpectedly.
+
+* __Payload:__ The payload contains structured information about a build event, encoded as a protocol buffer message specific to that event.
```

Comment:
Should we also mention that the payload is not necessarily the expected one, i.e., could also be an `Abort` message?